src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -98,6 +103,44 @@ public Parser getParser() {
         };
     }
 
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        if (StringUtil.isBlank(identifier)) {
+            return Optional.empty();
+        }
+
+        try (InputStream stream = new URLDownload(getURLForID(identifier)).asInputStream();
```

Comment:
What's the reason for these changes here? Maybe it's better to include them directly in IdBasedParserFetcher ?

Reply:
In all cases, there should be a comment on the byte reading etc. - So that the code is self-explanatory and no need to dive into the discussions in this PR.

Reply:
Changing JsonReader to handle empty input stream properly (as below), I will remove this from CrossRef.

Reply:
Fixed in https://github.com/mind000/jabref/pull/1.

Reply:
Hope that we did not do the same work in parallel ^^. Feel free to show up in Skype (I have the same id there) to discuss things. I think, this PR is close to be finished.